pkg/util/proxy/proxy.go

```diff
@@ -159,12 +171,33 @@ func Location(cluster *clusterapis.Cluster, tlsConfig *tls.Config) (*url.URL, ht
 		return nil, nil, err
 	}
 
-	proxyTransport, err := createProxyTransport(cluster, tlsConfig)
+	clusterHash := generateMd5HashForCluster(cluster)
+	if v, ok := clusterEndpointInfoStore.Load(clusterHash); ok {
+		clusterEndpointsInfo := v.(clusterEndpointInfo)
+		return location, clusterEndpointsInfo.Transport, nil
+	}
+
+	endpointInfo, err, _ := singleExecution.Do(clusterHash, func() (interface{}, error) {
+		if value, ok := clusterEndpointInfoStore.Load(clusterHash); ok {
+			return value, nil
+		}
+		proxyTransport, err := createProxyTransport(cluster, tlsConfig)
```

Comment:
What happens if tlsConfig changes?

Reply:
It will not  work fine,  cannot request successfully. But I think no need to consider the scenario abourt secrets  changes.